packages/tools/client-debugger/client-debugger-view/src/components/data-object-views/SharedMapView.tsx

```diff
@@ -30,6 +30,10 @@ export function SharedMapView(props: SharedMapViewProps): React.ReactElement {
 	const { sharedMap, renderChild } = props;
 
 	const [entries, setEntries] = React.useState<[string, unknown][]>([...sharedMap.entries()]);
+	const [collapsedEntries, setCollapsedEntries] = React.useState<{ [key: string]: boolean }>(
```

Comment:
Do we need to store this as React state? This is derived from `entries`, so we will always need to re-run this when that state changes anyways.

Reply:
Note that if we do keep this as React state, we need to update our `useEffect` block to re-run this whenever it updates `entries`, otherwise this state will potentially get out of sync with `entries`.